lib/process-services-cloud/src/lib/form/services/content-cloud-node-selector.service.ts

```diff
@@ -16,66 +16,90 @@
-    private dialog: MatDialog) {
-  }
+    sourceNodeNotFound = false;
 
-  openUploadFileDialog(currentFolderId?: string, selectionMode?: string, isAllFileSources?: boolean, restrictRootToCurrentFolderId?: boolean): Observable<Node[]> {
-    const select = new Subject<Node[]>();
-    select.subscribe({
-      complete: this.close.bind(this)
-    });
-    const data = <ContentNodeSelectorComponentData> {
-      title: 'Select a file',
-      actionName: NodeAction.ATTACH,
-      currentFolderId,
-      restrictRootToCurrentFolderId,
-      select,
-      selectionMode,
-      isSelectionValid: (entry: Node) => entry.isFile,
-      showFilesInResult: true,
-      showDropdownSiteList: false,
-      showLocalUploadButton: isAllFileSources
-  };
-    this.openContentNodeDialog(data, 'adf-content-node-selector-dialog', '66%');
-    return select;
-  }
+    constructor(
+        private apiService: AlfrescoApiService,
+        private logService: LogService,
+        private notificationService: NotificationService,
+        private dialog: MatDialog) {}
 
-    async fetchNodeIdFromRelativePath(alias: string, opts: { relativePath: string }): Promise<string> {
-        const relativePathNodeEntry: any = await this.apiService.getInstance().node
-        .getNode(alias, opts)
-        .catch((err) => this.handleError(err));
-        return relativePathNodeEntry?.entry?.id;
+    openUploadFileDialog(currentFolderId?: string, selectionMode?: string, isAllFileSources?: boolean, restrictRootToCurrentFolderId?: boolean): Observable<Node[]> {
+        const select = new Subject<Node[]>();
+        select.subscribe({
+            complete: this.close.bind(this)
+        });
+        const data = <ContentNodeSelectorComponentData> {
+            title: 'Select a file',
+            actionName: NodeAction.ATTACH,
+            currentFolderId,
+            restrictRootToCurrentFolderId,
+            select,
+            selectionMode,
+            isSelectionValid: (entry: Node) => entry.isFile,
+            showFilesInResult: true,
+            showDropdownSiteList: false,
+            showLocalUploadButton: isAllFileSources
+        };
+        this.openContentNodeDialog(data, 'adf-content-node-selector-dialog', '66%');
+        return select;
     }
 
-    async fetchAliasNodeId(alias: string): Promise<string> {
-        const aliasNodeEntry: any = await this.apiService.getInstance().node
-        .getNode(alias)
-        .catch((err) => this.handleError(err));
-        return aliasNodeEntry?.entry?.id;
+    fetchNodeIdFromRelativePath(alias: string, opts: { relativePath: string }): Promise<string> {
+        return from(this.apiService.nodesApi.getNode(alias, opts)).pipe(
+            map((relativePathNodeEntry: NodeEntry) => {
+                return relativePathNodeEntry.entry.id;
+            }),
+            catchError((error) => {
+                this.sourceNodeNotFound = true;
+                this.handleError(error);
+                return of(null);
```

Comment:
can we cahnge return of(null) with return of empty objects? {}

Reply:
@eromano @sivakumar414ram I had problems in the past with karma when returning nulls, so if some tests start failing, just try using empty object back